core/src/main/java/org/elasticsearch/Build.java

```diff
@@ -44,17 +48,26 @@
 
         Path path = getElasticsearchCodebase();
         if (path.toString().endsWith(".jar")) {
+            // The path might not be a .jar if we're running from an IDE
             try (JarInputStream jar = new JarInputStream(Files.newInputStream(path))) {
                 Manifest manifest = jar.getManifest();
-                shortHash = manifest.getMainAttributes().getValue("Change");
+                String change = manifest.getMainAttributes().getValue("Change");
+                shortHash = change == null ? "N/A" : change;
                 date = manifest.getMainAttributes().getValue("Build-Date");
             } catch (IOException e) {
                 throw new RuntimeException(e);
             }
         } else {
-            // not running from a jar (unit tests, IDE)
```

Comment:
Whoops - I didn't mean to flip this comment to line 51. Just an artifact of reverting most of a previous change. I'll fix that.